File: solutions/javascript/space-age/1/space-age.js
const planets = {
	MERCURY: "mercury",
	VENUS: "venus",
	EARTH: "earth",
	MARS: "mars",
	JUPITER: "jupiter",
	SATURN: "saturn",
	URANUS: "uranus",
	NEPTUNE: "neptune",
}

const EARTH_YEAR_IN_SECONDS = 31557600;

const orbitalPeriods = {
	[planets.MERCURY]: 0.2408467,
	[planets.VENUS]: 0.61519726,
	[planets.EARTH]: 1,
	[planets.MARS]: 1.8808158,
	[planets.JUPITER]: 11.862615,
	[planets.SATURN]: 29.447498,
	[planets.URANUS]: 84.016846,
	[planets.NEPTUNE]: 164.79132,
};

export function age(planet, seconds) {
	const earthYears = seconds / EARTH_YEAR_IN_SECONDS;
	const planetYears = earthYears / orbitalPeriods[planet];
	return Number(planetYears.toFixed(2));
}

File: solutions/scala/space-age/1/src/main/scala/SpaceAge.scala
object SpaceAge {
  private val EarthYearInSeconds = 31557600.0
  
  private val OrbitalPeriods = Map(
    "Earth" -> 1.0,
    "Venus" -> 0.61519726,
    "Mercury" -> 0.2408467,
    "Mars" -> 1.8808158,
    "Jupiter" -> 11.862615,
    "Saturn" -> 29.447498,
    "Uranus" -> 84.016846,
    "Neptune" -> 164.79132
  )

  def on(planet: String)(age: Double): Double = {
    val orbitalPeriod = OrbitalPeriods.getOrElse(planet, throw new IllegalArgumentException(s"Unknown planet: $planet"))
    (age / EarthYearInSeconds) / orbitalPeriod
  }

  def onEarth(age: Double): Double = on("Earth")(age)
  def onVenus(age: Double): Double = on("Venus")(age)
  def onMercury(age: Double): Double = on("Mercury")(age)
  def onMars(age: Double): Double = on("Mars")(age)
  def onJupiter(age: Double): Double = on("Jupiter")(age)
  def onSaturn(age: Double): Double = on("Saturn")(age)
  def onUranus(age: Double): Double = on("Uranus")(age)
  def onNeptune(age: Double): Double = on("Neptune")(age)
}